### ISSUE TITLE
Multi cursor deselects when typing brackets in multiple positions on same line

### ISSUE THREAD

**Issue by carlevans719** [Microsoft/vscode#17870]
- VSCode Version:
```
Version 1.8.1
Commit ee428b0eead68bf0fb99ab5fdc4439be227b6281
Date 2016-12-19T14:41:20.664Z
Shell 1.4.6
Renderer 53.0.2785.143
Node 6.5.0
```
- OS Version:
`Ubuntu 16.04  Kernel: 4.4.0-47-generic`

Steps to Reproduce:

1. Open a file - I have reproduced with `.less`, `.js` and `.html`
2. Select somewhere
3. Hold `alt` and select somewhere else further along the same line
4. Open and close a pair of brackets
5. Notice that the second pair (and all other subsequent pairs) of brackets did not close and the cursor is now only selecting the left-most place on the line.

The issue does *not* occur when typing only a closing bracket (unless that closing bracket is immediately after an opening one)

I've tried this with extensions both enabled and disabled - the issue is present in both situations

There are no errors in the dev tools console

**Comment by carlevans719** [comment]
Update: 
- It's not the left-most cursor that remains, it is the first/primary cursor.
- This does not happen when both cursors are inside a string in a `.js` file
- This did not happen in 1.6.0, but does happen in 1.8.1

**Comment by Kerollmops** [comment]
This seems to be fixed in v1.9, no ?

**Comment by carlevans719** [comment]
Confirmed: I no longer see the issue after upgrading to v1.9.0.
